engine/src/main/java/org/terasology/network/BroadCastServer.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.network;
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BroadCastServer {
+
+    private static DatagramSocket receiveSocket;
+    private static DatagramSocket sendSocket;
+    private static boolean turnOnBroadcast;
+    private static final Logger logger = LoggerFactory.getLogger(BroadCastServer.class);
+    private static ScheduledExecutorService service;
+    private static final String DISCOVERY_REQUEST = "DISCOVERY_REQUEST";
+    private static final String DISCOVERY_RESPONSE = "DISCOVERY_RESPONSE";
+
+    public boolean isBroadCastTurnedOn() {
+        return turnOnBroadcast;
+    }
+
+    public void startBroadcast() {
+        BroadCastServer.turnOnBroadcast = true;
+        try {
+            sendBroadCast();
+            Executors.newSingleThreadExecutor().execute(new Runnable() {
+                @Override
+                public void run() {
+                    while (true) {
+                        try {
+                            while (turnOnBroadcast) {
+                                listenToBroadCast();
+                            }
+                        } catch (Exception e) {
+                            logger.error("Broadcasting has been interrupted" + e.getMessage());
+                        }
+                    }
+                }
+            });
```

Comment:
1. Every invoke ```startBroadcast()``` will produce new unclosable Thread within ```newSingleThreadExecutor```
    With '2'(see next) you can eat all your CPU(or problem with shared access to listening socket), if you will click many times on button.
2. Using ```while (true)``` -  bad idea. It's Busy waiting (I googled it now. it said - anti-pattern :) ) 
3. You don't close sockets, listening socket will be listen while it not recieve packet or program exited, even  your turn broadcast off. 


I see two ways to fix it all:
1. working with something like ```UDPThread```(```extends Thread```) with closing socket on interrupt [how there](https://stackoverflow.com/a/4671080)
    create and interrupt it on start and stop broadcast
2.  not create new executor every time 
     instead ```while (true)``` use one of ```java.util.concurrent.*``` classes or ```Object#wait``` and ```Object#notify``` primitives
    setting timeouts for listening (avoid problem in '3')